src/step_definitions/performance.js

```diff
@@ -43,7 +45,36 @@ defineSupportCode(function ({ When }) {
     });
 
     return browser.driver.wait(() => {
-      return proxyDone;
+      return (
+        this.performanceReportFile = uniqueFileName,
+        proxyDone
+      )
     });
   });
+
+  Then(/^the requests should take a maximum of "([^"]*)" milliseconds$/, function (maxTiming) {
+    const performanceReport = JSON.parse(fs.readFileSync(`reports/performance/${this.performanceReportFile}`, 'utf8'));
+    const mappedRequests = performanceReport.log.entries.map(item => ({
+      ttfb: item.timings.wait,
+      url: item.request.url
+    }));
+
+    if (mappedRequests.length > 0) {
+      const slowRequests = mappedRequests.filter(request => request.ttfb > maxTiming);
+      let reportList = [];
```

Comment:
Instead of variable, use slowRequests.forEach(function(item){...});  and  just console.log it and do reject at the end. There is absolutely no reason to push it to table.

Reply:
Improved.